[docs] Add simple transition guide (#579)
membraneframework/live_compositor: docs/pages/guides/view-transition.md

          children: [
            %{
              type: :rescaler,
              id: "rescaler_1",
              // highlight-start
              width: 480,
              // highlight-end
              child: %{ type: :input_stream, input_id: :input_1 },
            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        start_timer: {:add_videos_timer, Membrane.Time.seconds(2)}
      ], state}
    end

    def handle_tick(:start_transition, _ctx, state)
      request = %LiveCompositor.Request.UpdateVideoOutput{
        output_id: "output_1",
        root: %{
          type: :view,
          background_color_rgba: "#4d4d4dff",
          children: [
            %{
              type: :rescaler,
              id: "rescaler_1",
              // highlight-start
              width: 1280,
              transition: %{ duration_ms: 2000 },
              // highlight-end
              child: %{ type: :input_stream, input_id: :input_1 },
            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        stop_timer: :start_transition
      ], state}
    end
    ```
  </TabItem>
</Tabs>

> 

Author: That would fit better in a separate guides that handles scheduling and offline processing

> In the first update request you can see that rescaler has width of 480, and in the second one it is changed
> to 1280 and `transition.duration_ms: 2000` was added.
> 
> It is important that component has the same `"id"` in both initial state and the update that starts the
> transition, otherwise it will switch immediately to the new state without a transition.

Reviewer: ```suggestion
In the first update request, you can see that the rescaler has a width of 480, and in the second one, it is changed
to 1280 and `transition.duration_ms: 2000` was added.

The component must have the same `"id"` in both the initial state and the update that starts the
transition, otherwise it will switch immediately to the new state without a transition.
```

<div style={{textAlign: 'center'}}>
    <img src={SimpleTransition} style={{ width: 600 }} />
    Output stream
</div>

> ### Transition on one of the sibling components
> 
> In the above scenario you saw how transition on single component behave, but let's see what happens with
> components that are not a part of the transition, but their size and position still depends on other components.
> 
> Add second input stream wrapped with `Rescaler`, but without any transition options.

Reviewer: ```suggestion
### Transition on one of the sibling components

In the above scenario you saw how transition on a single component behaves, but let's see what happens with
components that are not a part of the transition, but their size and position still depend on other components.

Add a second input stream wrapped with `Rescaler`, but without any transition options.
```

            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        start_timer: {:add_videos_timer, Membrane.Time.seconds(2)}
      ], state}
    end

    def handle_tick(:start_transition, _ctx, state)
      request = %LiveCompositor.Request.UpdateVideoOutput{
        output_id: "output_1",
        root: %{
          type: :view,
          background_color_rgba: "#4d4d4dff",
          children: [
            %{
              type: :rescaler,
              id: "rescaler_1",
              // highlight-start
              width: 1280,
              transition: %{ duration_ms: 2000 },
              // highlight-end
              child: %{ type: :input_stream, input_id: :input_1 },
            },
            %{
              type: :rescaler,
              child: %{ type: :input_stream, input_id: :input_2 },
            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        stop_timer: :start_transition
      ], state}
    end
    ```
  </TabItem>
</Tabs>

<div style={{textAlign: 'center'}}>
    <img src={BothStreamsTransition} style={{ width: 600 }} />
    Output stream
</div>

> ### Transition between different modes
> 
> Currently, a state before the transition and after needs to use the same type of configuration. In particular:
> - It is not possible to transition a component between static and absolute positioning.
> - It is not possible to transition a component between using `top` and `bottom` fields (the same for `left`/`right`).
> - It is not possible to transition a component with known `width`/`height` to state with dynamic `width/height` based
> on parent layout.
> 
> Let's try the same example as in first scenario with a single input, but instead change `Rescaler` component to be absolutely positioned in the second update.

Reviewer: ```suggestion
### Transition between different modes

Currently, a state before the transition and after needs to use the same type of configuration. In particular:
- It is not possible to transition a component between static and absolute positioning.
- It is not possible to transition a component between using `top` and `bottom` fields (the same for `left`/`right`).
- It is not possible to transition a component with known `width`/`height` to a state with dynamic `width/height` based
on the parent layout.

Let's try the same example as in the first scenario with a single input, but instead, change the `Rescaler` component to be absolutely positioned in the second update.
```

              type: :rescaler,
              id: "rescaler_1",
              // highlight-start
              width: 480,
              // highlight-end
              child: %{ type: :input_stream, input_id: :input_1 },
            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        start_timer: {:add_videos_timer, Membrane.Time.seconds(2)}
      ], state}
    end

    def handle_tick(:start_transition, _ctx, state)
      request = %LiveCompositor.Request.UpdateVideoOutput{
        output_id: "output_1",
        root: %{
          type: :view,
          background_color_rgba: "#4d4d4dff",
          children: [
            %{
              type: :rescaler,
              id: "rescaler_1",
              // highlight-start
              width: 1280,
              top: 0,
              left: 0,
              transition: %{ duration_ms: 2000 },
              // highlight-end
              child: %{ type: :input_stream, input_id: :input_1 },
            }
          ]
        }
      }
      {[
        notify_child: {:live_compositor, request},
        stop_timer: :start_transition
      ], state}
    end
    ```
  </TabItem>
</Tabs>

> As you can see on the resulting stream transition did not happen because the `Rescaler` component
> in the initial scene was using static positioning and after the update it was positioned absolutely.

Reviewer: ```suggestion
As you can see the resulting stream transition did not happen because the `Rescaler` component
in the initial scene was using static positioning and after the update it was positioned absolutely.
```

<div style={{textAlign: 'center'}}>
    <img src={UnsupportedTransition} style={{ width: 600 }} />
    Output stream
</div>

> ### Different interpolation functions
> 
> All of the above examples were using default linear interpolation, but there are also few other
> modes available.

Reviewer: ```suggestion
### Different interpolation functions

All of the above examples use default linear interpolation, but there are also a few other
modes available.
```
